File: uri/soma/soma.kt
import java.util.Scanner
import kotlin.text.*

fun soma( a: Int, b: Int ) : Int {
    return a + b
}

fun main( args: Array<String>)
{
    var input = Scanner(System.`in`)
    var a : Int;  var b : Int; var resultado : Int
    a = input.nextInt()
    b = input.nextInt()
    resultado = soma(a, b)
    println("SOMA = $resultado")
}

File: uri/soma/soma.hs
{-# LANGUAGE FlexibleInstances #-}

import Text.Printf
import System.IO

soma :: Integer -> Integer -> Integer
soma x y = x + y

main = do
    x <- readLn
    y <- readLn
    let resultado = soma x y :: Integer
    printf "SOMA = %d\n" resultado
